src/service/irc/IrcService.cpp

```diff
@@ -111,68 +111,68 @@ bool IrcService::onEvent(std::shared_ptr<IEvent> event) {
             lock_guard<mutex> lock(connection.getChannelLoginDataMutex());
             connection.removeHost(del->getHost(), del->getPort());
         }
-    } else if (type == EventQueryChats::uuid) {
-        auto query = event->as<EventQueryChats>();
+    } else if (type == EventQuery::uuid) {
+        auto query = event->as<EventQuery>();
+        if (query->getType() == EventQueryType::Chats) {
+            size_t firstId = IdProvider::getInstance().getLastId("irc_log"); // for backlog requests
+            auto listing = make_shared<EventIrcChatListing>(firstId, userId, query->getData());
 
-        size_t firstId = IdProvider::getInstance().getLastId("irc_log"); // for backlog requests
-        auto listing = make_shared<EventIrcChatListing>(firstId, userId, query->getData());
-
-        std::list<lock_guard<mutex>> locks;
-        // lock all to assure correct results
-        for (auto& cxnPair : ircConnections) {
-            auto& connection = cxnPair.second;
-            locks.emplace_back(connection.getChannelLoginDataMutex());
-        }
+            std::list<lock_guard<mutex>> locks;
+            // lock all to assure correct results
+            for (auto& cxnPair : ircConnections) {
+                auto& connection = cxnPair.second;
+                locks.emplace_back(connection.getChannelLoginDataMutex());
+            }
 
-        for (auto& cxnPair : ircConnections) {
-            auto& connection = cxnPair.second;
-            IrcServerListing& server = listing->addServer(connection.getActiveNick(),
-                                                          connection.getServerId(),
-                                                          connection.getServerName());
-            for (auto& channelPair : connection.getChannelStore()) {
-                string channelName = channelPair.first;
-                const IrcChannelStore& channelStore = channelPair.second;
-                IrcChannelListing& channel = server.addChannel(channelName,
-                                                               channelStore.getTopic(),
-                                                               channelStore.getDisabled());
-                for (auto& userPair : channelStore.getUsers()) {
-                    string username = userPair.second.getNick();
-                    channel.addUser(username, "");
+            for (auto& cxnPair : ircConnections) {
+                auto& connection = cxnPair.second;
+                IrcServerListing& server = listing->addServer(connection.getActiveNick(),
```

Comment:
`EventIrcChatListing` has `addServer` method with three arguments, but `addServer` method in `EventIrcSettingsListing` has two arguments - without `activeNick`. There is also different result type in both.

I could try to make one **template class** for `EventIrc{Settings,Chat}Listing` with listing parameter, but it will probably cause further changes in classes underneath (like `IrcServerListing`). What do you think about that?